clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -10426,46 +10426,15 @@ static void handleFunctionReturnThunksAttr(Sema &S, Decl *D,
   D->addAttr(FunctionReturnThunksAttr::Create(S.Context, Kind, AL));
 }
 
-static constexpr std::pair<Decl::Kind, StringRef>
-MakeDeclContextDesc(Decl::Kind K, StringRef SR) {
-  return std::pair<Decl::Kind, StringRef>{K, SR};
-}
-
-// FIXME: Refactor Util class in SemaSYCL.cpp to avoid following
-// code duplication.
 bool isDeviceAspectType(const QualType Ty) {
   const EnumType *ET = Ty->getAs<EnumType>();
   if (!ET)
     return false;
 
-  std::array<std::pair<Decl::Kind, StringRef>, 3> Scopes = {
-      MakeDeclContextDesc(Decl::Kind::Namespace, "sycl"),
-      MakeDeclContextDesc(Decl::Kind::Namespace, "_V1"),
-      MakeDeclContextDesc(Decl::Kind::Enum, "aspect")};
-
-  const auto *Ctx = cast<DeclContext>(ET->getDecl());
-  StringRef Name = "";
-
-  for (const auto &Scope : llvm::reverse(Scopes)) {
-    Decl::Kind DK = Ctx->getDeclKind();
-    if (DK != Scope.first)
-      return false;
+  if (const SYCLTypeAttr *Attr = ET->getDecl()->getAttr<SYCLTypeAttr>())
```

Comment:
Nit: use auto here?

Reply:
I can make these changes but I was wondering if this there are any guidelines for this. Does community prefer we use `auto `in such cases?

Reply:
The guidelines say this: 
'Don’t “almost always” use auto, but do use auto with initializers like cast<Foo>(...) or other places where the type is already obvious from the context.'

In this case, the type is obvious from the getAttr<...> context, but I leave it up to you.   We are inconsistent in this file (we use both auto and explicit type in multiple places), and even within this change-set, we are.  :-)

(Except for that auto/auto * case that I pointed out below.   That needs to be made.  Guidelines are clear on that.)

Reply:
Thanks for the info @premanandrao. I made the changes. I'm building locally to confirm nothing broke. I will upload changes soon.